### PR TITLE
Remove a few linters from efficiency tag

### LINKED ISSUE #2653: Revisit linters included in the list of efficiency linters?
I am surprised by some linters included under efficiency tag.

``` r
library(lintr)
names(linters_with_tags("efficiency"))
#>  [1] "any_duplicated_linter"            "any_is_na_linter"                
#>  [3] "boolean_arithmetic_linter"        "consecutive_mutate_linter"       
#>  [5] "fixed_regex_linter"               "if_switch_linter"                
#>  [7] "ifelse_censor_linter"             "inner_combine_linter"            
#>  [9] "length_test_linter"               "lengths_linter"                  
#> [11] "list_comparison_linter"           "literal_coercion_linter"         
#> [13] "matrix_apply_linter"              "nested_ifelse_linter"            
#> [15] "nrow_subset_linter"               "nzchar_linter"                   
#> [17] "outer_negation_linter"            "redundant_equals_linter"         
#> [19] "redundant_ifelse_linter"          "regex_subset_linter"             
#> [21] "routine_registration_linter"      "sample_int_linter"               
#> [23] "scalar_in_linter"                 "seq_linter"                      
#> [25] "sort_linter"                      "string_boundary_linter"          
#> [27] "undesirable_function_linter"      "undesirable_operator_linter"     
#> [29] "unnecessary_concatenation_linter" "unnecessary_lambda_linter"       
#> [31] "vector_logic_linter"              "which_grepl_linter"
```

<sup>Created on 2024-09-05 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

---

IMO, the following linters shouldn't be in this category:

- `undesirable_function_linter()` and `undesirable_operator_linter()` (which are more about best practices)
- `list_comparison_linter()` suggests code that is actually slower

``` r
x <- list(c(1, 2, 3), c(4, 5, 6), c(7, 8, 9))

bench::mark(
  lapply(x, sum) > 10,
  unlist(lapply(x, sum)) > 10,
  iterations = 1000L
)
#> # A tibble: 2 × 6
#>   expression                       min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>                  <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 lapply(x, sum) > 10           1.31µs   1.48µs   631356.        0B        0
#> 2 unlist(lapply(x, sum)) > 10   1.68µs   1.93µs   437505.        0B        0
```

<sup>Created on 2024-09-05 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

---

WDYT? Maybe you also a few others that in hindsight stick out.

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
closes #2653

### CODE REVIEW

**Comment by MichaelChirico** [comment]
I don't think we even need a NEWS item TBH.

**Comment by AshesITR** [comment]
Technically, it can break user configs if they use the efficiency tag, so I think it's worth mentioning. The blast radius of this change is most likely miniscule, though.

File: NEWS.md [hunk 1]
```diff
@@ -18,6 +18,9 @@
 * `cyclocomp_linter()` is no longer part of the default linters (#2555, @IndrajeetPatil) because the tidyverse style guide doesn't contain any guidelines on meeting certain complexity requirements. Note that users with `cyclocomp_linter()` in their configs may now need to install {cyclocomp} intentionally, in particular in CI/CD pipelines.
 * `scalar_in_linter` is now configurable to allow other `%in%` like operators to be linted. The data.table operator `%chin%` is no longer linted by default; use `in_operators = "%chin%"` to continue linting it. (@F-Noelle)
 * `lint()` and friends now normalize paths to forward slashes on Windows (@olivroy, #2613).
+* Following linters have been removed from efficiency linters (@IndrajeetPatil, #2655):
+  `undesirable_function_linter()`, `undesirable_operator_linter()`, and `list_comparison_linter()`
```
[the remark this answers is not in the document]
Comment (MichaelChirico) [reply NEWS.md:22]: I think here is best

Comment (MichaelChirico) [inline NEWS.md:22]:
```suggestion
* `undesirable_function_linter()`, `undesirable_operator_linter()`, and `list_comparison_linter()` were removed from the tag `efficiency` (@IndrajeetPatil, #2655). If you use `linters_with_tags("efficiency")` to include these linters, you'll need to adjust your config to keep linting your code against them. We did not find any such users on GitHub.
```

Comment (MichaelChirico) [reply NEWS.md:22]: Ref: https://github.com/search?q=lang%3AR+%2Flinters_with_tags%5B%28%5D%2F+%2Fefficiency%2F+-path%3Alinter_tag_docs.R&type=code

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**